### ISSUE TITLE
emerge rstudio fails if no hunspell dictionaries are installed

### ISSUE THREAD

**Issue by chris2b5** [jgenco/jgenco-overlay#10]
If hunspell is not installed (or if it is already installed but without dictionaries) there may be no `/usr/share/hunspell` and the ebuild will fail in one of these places:

[rstudio-9999.ebuild#L360](https://github.com/jgenco/jgenco-overlay/blob/bb489c0d3d1f13afac245f6f359f75284468711f/sci-mathematics/rstudio/rstudio-9999.ebuild#L360)
[rstudio-2022.12.0.353.ebuild#L335](https://github.com/jgenco/jgenco-overlay/blob/bb489c0d3d1f13afac245f6f359f75284468711f/sci-mathematics/rstudio/rstudio-2022.12.0.353.ebuild#L335)
[rstudio-2023.03.0.386.ebuild#L362](https://github.com/jgenco/jgenco-overlay/blob/bb489c0d3d1f13afac245f6f359f75284468711f/sci-mathematics/rstudio/rstudio-2023.03.0.386.ebuild#L362)

Like this:

```
>>> Unpacking rstudio-2020.12.0.353.tar.gz to /var/tmp/portage/sci-mathematics/rstudio-2022.12.0.353/work
 * ERROR: sci-mathematics/rstudio-2022.12.0.353::jgenco failed (unpack phase):
 *   /usr/share/hunspell not a directory of file
 * 
 * Call stack:
 *     ebuild.sh, line  136:  Called src_unpack
 *   environment, line 5114:  Called bundle_ln '/usr/share/hunspell' '/dependencies/dictionaries' 'dictionaries'
 *   environment, line 1228:  Called die
 * The specific snippet of code:
 *       [[ ! -d "${EPREFIX}${1}" && ! -f "${EPREFIX}${1}" ]] && die "${EPREFIX}${1} not a directory of file";
```

It is easy for `hunspell` to get installed without dictionaries, but it is also looks easy to cause dictionaries to get installed when installing `hunspell`.

Best approach is probably for the user to set the `USE_EXPAND` variable `L10N` in their `/etc/portage/make.conf` per the instructions in the [Localization Guide](https://wiki.gentoo.org/wiki/Localization/Guide#L10N). But if they have not done that then installing hunspell will not install dictionaries and the rstudio install will fail as above.

Not sure what the best approach is to address this here. Some ideas:
1. give more informative error message when `/usr/share/hunspell` is missing 
2. somehow require the user to have set `L10N` prior to installing `hunspell` 
3. brute force: if `L10N` is not set, we simply set it to some reasonable default like `L10N="en"` prior to installing `hunspell`, or equivalently we set `l10n_en` on `hunspell`

`hunspell` needs to get emerged with `L10N` (or use `l10n_en`) set in order for it to pull in the dictionaries and create `/usr/share/hunspell`, even if `hunspell` was already installed but with no dictionaries... 

It is also possible to install dictionary packages directly e.g. `app-dicts/myspell-en` but probably better when the localisation is set up correctly (and this still has the issue of installing e.g. the `en` dictionaries which the user may not want)?

[nb thanks very much for this repo, rstudio working nicely for me now :-) ]

**Comment by chris2b5** [comment]
Thanks, I understand this better now. 

I agree, a warning if `/usr/share/hunspell` is missing is probably best - at least for now / as long as the worst that happens is that Spell Check silently fails. Nice and simple, and it's not obvious what else we can do that would be better.

If I have some time, I'll look at other packages using `hunspell` to see if there is a solution we can use. Likely we're not the first to run into this.

**Comment by jgenco** [comment]
`rg -i "dictionar(y|ies)" /var/db/repos/gentoo/ -tebuild -tdiff` produced some hits.
like [media-video/gaupol](https://github.com/gentoo/gentoo/blob/7ac919e7c9d468ae5c32ab4ba259e5da86da4335/media-video/gaupol/gaupol-1.11-r1.ebuild#L43-L59) and [app-text/nuspell](https://github.com/gentoo/gentoo/blob/7ac919e7c9d468ae5c32ab4ba259e5da86da4335/app-text/nuspell/nuspell-5.1.2.ebuild#L43-L51)
Both put the warning in pkg_postinst instead of src_install like I previously did.
I was thinking  for the warning something like this:
```bash
    if [[ ! -d "${EPREFIX}/usr/share/hunspell" ]];then
        elog "RStudio's spell check needs at least one"
        elog "app-dicts/myspell-* dictionary to be installed."
        elog ""
        elog "or set the L10N variable in /etc/poratage/make.conf"
        elog "see https://wiki.gentoo.org/wiki/Localization"
    fi
```

**Comment by chris2b5** [comment]
It looks like you had more luck than I did. 

I looked at a bunch of packages that had dependencies on `hunspell` but found that almost all didn't handle the problem at all. There were a couple of exceptions:

* [libreoffice-l10n](https://gitweb.gentoo.org/repo/gentoo.git/tree/app-office/libreoffice-l10n/libreoffice-l10n-7.5.2.1.ebuild) (which gets pulled in by the behemoth that is `libreoffice`) appears to want to install all the world's dictionaries.
* [app-emacs/markdown-mode](https://gitweb.gentoo.org/repo/gentoo.git/tree/app-emacs/markdown-mode/markdown-mode-2.5-r1.ebuild) looks like it defaults to the `en` dicts:

```bash
IUSE="test"
RESTRICT="!test? ( test )"
...
...
BDEPEND="test? (
	virtual/pandoc
	|| ( app-text/aspell[l10n_en] app-text/hunspell[l10n_en] )
)"
```

None of which really helps.

Long story short, your plan seems good to me!

**Comment by jgenco** [comment]
Thanks, for the report.
FWI: I only updated rstudio-2023.03.0-386 as rstudio-2022.12.0.353 is now an old version and RStudio-9999 will be updated later.
